ground/src/main/java/com/google/android/ground/ui/map/LoiController.kt

```diff
@@ -30,27 +28,23 @@ import io.reactivex.Flowable
 import io.reactivex.subjects.PublishSubject
 import io.reactivex.subjects.Subject
 import javax.inject.Inject
+import javax.inject.Singleton
 
-/** Provides data for displaying cards for visible LOIs at the bottom of the screen. */
-class LoiCardSource
+@Singleton
+class LoiController
 @Inject
 internal constructor(
   private val surveyRepository: SurveyRepository,
   private val locationOfInterestRepository: LocationOfInterestRepository,
 ) {
 
   private val cameraBoundsSubject: @Hot Subject<LatLngBounds> = PublishSubject.create()
-  val locationsOfInterest: LiveData<List<LocationOfInterest>>
 
-  init {
+  fun loisWithinMapBounds(): Flowable<List<LocationOfInterest>> {
```

Comment:
Can this be moved into the loiRep as `getWithinBounds(LatLngBounds)`? The camera bounds should probably live in the `MapStateRepository` or `BaseMapViewModel`, then this class can be deleted.